### PR TITLE
[bot] treewide: remove unreferenced code

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

 remove unreferenced code

### CODE REVIEW

**Comment by zimbatm** [comment]
Nice. What methodology did you use to detect unreferenced code?

File: pkgs/applications/altcoins/mist.nix [hunk 1]
```diff
@@ -6,14 +6,6 @@ let
 
   throwSystem = throw "Unsupported system: ${stdenv.system}";
 
-  meta = with stdenv.lib; {
```
Comment (orivej) [inline pkgs/applications/altcoins/mist.nix]: This should be used later in buildFHSUserEnv and probably mkDerivation.

**Comment by volth** [comment]
> What methodology did you use to detect unreferenced code?

A simple query against the AST: values declared in ```let``` but not used in ```in```.
So, it is not complete, the should be much more unreferenced code, not in ```let```-blocks

**Comment by oxij** [comment]
Cool! LGTM.

There're unused things that look useful to me, but I'm ok with dropping them, people can just revert a piece of this if they happen to need them later.

Meanwhile, I found issues in several precious to me things by looking at this.

1) Unused `markBroken` in `org-packages.nix` is duplicated in sibling files, it's an issue with Emacs infra, IMHO.
   /cc @ttuegel @mdorman

2) Unused `xsaPatch` in `xen` is a left-over from XSA rework which was not applied to 4.5.
   4.5 just needs to be dropped. Will do myself later.

**Comment by volth** [comment]
> do you mind sharing the code that you used to generate this? 

@zimbatm https://gist.github.com/volth/297b24d577f9868255210c31479fd56b

> Without applying it regularly against nixpkgs the code base will probably diverge again. 

I think it would be better to implement warnings in Nix (about unused code/vars, whitespace at the line end, tabs, ...) so people get notified about code problems as they write the code.
And ofborg could print warnings found in PRs

**Comment by ben0x539** [comment]
Hey y'all I think this PR breaks the build for me with

> error: undefined variable 'registryConfig' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/misc/docker-registry.nix:8:94

Evaluation of my system config succeeds with ce12ad31169, but fails with 87f5930c3fb. I'm working around this by removing `services.dockerRegistry.enable = true;` from my config, which I don't really use anyway.

I'm not quite sure where that variable is supposed to be coming from, exactly, is this possibly down to me doing anything funny on my end or am I just being outsmarted by the docker registry module?

**Comment by volth** [comment]
> Evaluation of my system config succeeds with ce12ad3, but fails with 87f5930

what an error does it fail with?

**Comment by ben0x539** [comment]
> what an error does it fail with?

> error: undefined variable 'registryConfig' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/misc/docker-registry.nix:8:94

**Comment by volth** [comment]
@ben0x539 fixed in https://github.com/NixOS/nixpkgs/pull/44546

**Pull request merged**